packages/react/src/ActionList/Item.tsx

```diff
@@ -219,18 +260,46 @@ const UnwrappedItem = <As extends React.ElementType = 'li'>(
     id: itemId,
   }
 
-  const containerProps = _PrivateItemWrapper
-    ? {role: itemRole ? 'none' : undefined, ...props}
-    : // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-      (listSemantics && {...menuItemProps, ...props, ref: forwardedRef}) || {}
+  // Props distribution depends on which component pattern is being used:
+  // 1. _PrivateItemWrapper - used by LinkItem
+  // 2. isLinkItem - direct Item usage with link props
+  // 3. default - regular button/div behavior
+
+  // Create a Set for lookup of interactive-only props
+  const interactivePropsSet = new Set<string>(INTERACTIVE_ELEMENT_PROPS)
+
+  // Split props into container and interactive element props
+  const containerOnlyProps: Record<string, unknown> = {}
+  const interactiveProps: Record<string, unknown> = {}
+
+  for (const [key, value] of Object.entries(props)) {
+    if (interactivePropsSet.has(key)) {
```

Comment:
Sorry, just curious, is there a benefit of converting the array into a set to use Set.has instead of Array.includes? Took me a second to follow the trail 😅